frontend/model/contracts/group.js

```diff
@@ -313,38 +325,43 @@ sbp('chelonia/defineContract', {
       return getters.groupSettings.mincomeCurrency
     },
     periodStampGivenDate (state, getters) {
-      return (recentDate: string | Date) => {
-        if (typeof recentDate !== 'string') {
-          recentDate = recentDate.toISOString()
+      return (date: Date | string): string | null => {
+        const givenStamp = typeof date === 'string' ? date : date.toISOString()
+        // Should we throw instead?
+        if (!isPeriodStamp(givenStamp)) return null
```

Comment:
Yes, please throw `TypeError('must be periodstamp')` (note: purposefully not localized because too obscure and developer-facing)